lorawan-encoding/src/securityhelpers.rs

```diff
@@ -58,17 +56,16 @@ pub fn encrypt_frm_data_payload(
     generate_helper_block(phy_payload, 0x01, fcnt, &mut a[..]);
 
     let mut s = [0u8; 16];
-    let s_block = GenericArray::from_mut_slice(&mut s[..]);
 
     let mut ctr = 1;
     for i in 0..len {
         let j = i & 0x0f;
         if j == 0 {
             a[15] = ctr;
             ctr += 1;
-            s_block.copy_from_slice(&a);
```

Comment:
`s = a` means we can just use `a` and skip `s` entirely?

Reply:
The problem would be that we would re-encrypt the "helper_block" ie `a` on long messages. I tried it and it makes the "test_complete_long_dataup_payload_frm_payload" test fail which makes sense.